velox/exec/MergeSource.cpp

```diff
@@ -17,25 +17,79 @@
 #include "velox/exec/MergeSource.h"
 
 #include <boost/circular_buffer.hpp>
+#include "velox/common/testutil/TestValue.h"
 #include "velox/exec/Merge.h"
 #include "velox/vector/VectorStream.h"
 
+using facebook::velox::common::testutil::TestValue;
+
 namespace facebook::velox::exec {
 namespace {
 
+namespace {
+class PromiseNotifier {
+ public:
+  explicit PromiseNotifier(size_t initSize) {
+    promises_.reserve(initSize);
+  }
+
+  ~PromiseNotifier() {
```

Comment:
dtor relied mechanism. Shall we call it Scoped... to align with rest of the codebase?

Reply:
updated.